tests/test_exchange_collector.py

```diff
@@ -0,0 +1,139 @@
+    Then it is called several times (at time 8 and time 23) and it should write the value of example_flux_output
+    at those times to netcdf
+    """
+
+    if not (os.path.isdir(tmp_path_dev)):
+        os.makedirs(tmp_path_dev)
+
+    config_dict = tomli.loads(output_config_toml)
+    config_dict["general"]["output_dir"] = tmp_path_dev
+    exchange_collector = ExchangeCollector(config_dict)
+
+    some_array0: NDArray[np.float_] = NDArray[np.float_](
+        (5,), buffer=np.array([1.1, 2.0, -4.8, np.nan, 3])
+    )
+    some_array1: NDArray[np.float_] = NDArray[np.float_](
+        (5,), buffer=np.array([1.1, 666.0, -4.8, 0.0, 3])
+    )
+    exchange_collector.log_exchange("example_flux_output", some_array0, 8.0)
+    exchange_collector.log_exchange("example_flux_output", some_array1, 23.0)
+    exchange_collector.finalize()
+
+    # read the file it just wrote and check the contents
+    ds = nc.Dataset(tmp_path_dev / "example_flux_output.nc", "r")
+    dat = ds.variables["xchg"][:]
+    tim = ds.variables["time"][:]
+    assert np.array_equal(dat[0, :], some_array0, equal_nan=True)
+    assert np.array_equal(dat[1, :], some_array1, equal_nan=True)
+    assert np.array_equal(tim[:], np.array([8.0, 23.0]), equal_nan=True)
+
+
+def test_exchange_collector_ignores_unknown_exchanges(
+    tmp_path_dev: Path, output_config_toml: str
+) -> None:
+    """
+    The exchange colector should ignore arrays that it is not asked to log
+    """
+    config_dict = tomli.loads(output_config_toml)
+    config_dict["general"]["output_dir"] = tmp_path_dev
+    exchange_collector = ExchangeCollector(config_dict)
+
+    some_array0: NDArray[np.float_] = NDArray[np.float_](
+        (5,), buffer=np.array([1.1, 2.0, -4.8, np.nan, 3])
+    )
+    some_array1: NDArray[np.float_] = NDArray[np.float_](
+        (5,), buffer=np.array([1.1, 666.0, -4.8, 0.0, 3])
+    )
+    exchange_collector.log_exchange("non_existing_type", some_array0, 8.0)
+    exchange_collector.log_exchange("non_existing_type", some_array1, 23.0)
+    exchange_collector.finalize()
+
+
+def test_exchange_collector_raises_exception_when_array_size_varies(
+    tmp_path_dev: Path, output_config_toml: str
+) -> None:
+    """
+    While an array can vary through time, its dimensions should not.
+    """
+    config_dict = tomli.loads(output_config_toml)
+    config_dict["general"]["output_dir"] = tmp_path_dev
+    exchange_collector = ExchangeCollector(config_dict)
+
+    some_array: NDArray[np.float_] = NDArray[np.float_](
+        (5,), buffer=np.array([1.1, 2.0, -4.8, np.nan, 3])
+    )
+    some_smaller_array: NDArray[np.float_] = NDArray[np.float_](
+        (4,), buffer=np.array([1.1, 666.0, -4.8, 0.0])
+    )
+    exchange_collector.log_exchange("example_stage_output", some_array, 8.0)
+    with pytest.raises(ValueError) as e:
+        exchange_collector.log_exchange(
+            "example_stage_output", some_smaller_array, 23.0
+        )
+    assert (
```

Comment:
That can be done within `pytest.raises` with a regex expression

https://docs.pytest.org/en/4.6.x/reference.html#pytest-raises

Reply:
done